src/spikeinterface/preprocessing/decimate.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+from spikeinterface.core.core_tools import (
+    define_function_from_class,
+)
+
+from .basepreprocessor import BasePreprocessor
+from .filter import fix_dtype
+from ..core import BaseRecordingSegment
+
+
+class DecimateRecording(BasePreprocessor):
+    """
+    Decimate the recording extractor traces using array slicing
+
+    Important: This uses simple array slicing for decimation rather than eg scipy.decimate.
+    This might introduce aliasing, or skip across signal of interest.
+    Consider  spikeinterface.preprocessing.ResampleRecording for safe resampling.
+
+    Parameters
+    ----------
+    recording : Recording
+        The recording extractor to be decimated.
+    decimation_factor : int
+        Step between successive frames sampled from the parent recording.
+    decimation_offset : int, default: 0
+        Index of first frame sampled from the parent recording.
+        Expecting `decimation_offset` < `decimation_factor`, and `decimation_offset` < parent_recording.get_num_samples()
+        to ensure that the decimated recording has at least one frame. Consider combining DecimateRecording
+        with FrameSliceRecording for fine control on the recording start and end frames.
+
+    Returns
+    -------
+    decimate_recording: DecimateRecording
+        The decimated recording extractor object. The full traces of the child recording correspond
+        to the parent traces as follows:
+            ```<decimated_traces> = <parent_traces>[<decimation_offset>::<decimation_factor>]```
+
+    """
+
+    name = "decimate"
+
+    def __init__(
+        self,
+        recording,
+        decimation_factor,
+        decimation_offset=0,
+    ):
+        # Original sampling frequency
+        self._orig_samp_freq = recording.get_sampling_frequency()
+        parent_nsamp = recording.get_num_samples()
```

Comment:
```suggestion
        parent_nsamp = min([recording.get_num_samples(segment_index) for segment_index in range(recording.get_num_segments()])
```
Parent can be multi-segment! ;)

Reply:
OK so should what should we do for multi-segments...
Same decimation factor for all segments, and accept decimation_offset to be either an int or an array of ints ?

Reply:
I think same for both is fine

Reply:
ok done